File: Blog.Web/ClientApp/src/app/core/models/Post.ts
import { User } from './User';

/**
 * Post model.
 */
export class Post {
  /*
  //access -> string
  */
  /**
   * Post Id
   * @param id number
   * Post Title
   * @param title string
   * Post Description
   * @param description string
   * Post Content
   * @param content string
   * Post Author Id
   * @param authorId number
   * Post Author
   * @param author User
   * Post Seen
   * @param seen number
   * Post Likes
   * @param likes number
   * Post Dislikes
   * @param dislikes number
   * Post Image Url
   * @param imageUrl string
   * Post Tags
   * @param tags string
   * Post Tags List
   * @param tagsList string
   * Post Comments
   * @param comments array
   * Post Comments Count
   * @param commentsCount number
   * Post Created at
   * @param createdAt string
   */
  constructor (
    public id: number,
    public title: string,
    public description: string,
    public content: string,
    public authorId: string,
    public author: User,
    public seen: number,
    public likes: number,
    public dislikes: number,
    public imageUrl: string,
    public tags: string,
    public createdAt: string,
    public commentsCount: number,
    public comments?: Comment[],
    public tagsList?: string[],
  ) {}
}

File: Blog.Data.Models/Post.cs
using System.Collections.Generic;

namespace Blog.Data.Models
{
    using System;
    using Core;

    /// <summary>
    /// Post model.
    /// </summary>
    public class Post : Entity
    {
        /// <summary>
        /// Gets or sets title.
        /// </summary>
        public string Title { get; set; }

        /// <summary>
        /// Gets or sets description.
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Gets or sets content.
        /// </summary>
        public string Content { get; set; }

        /// <summary>
        /// Gets or sets seen.
        /// </summary>
        public int Seen { get; set; }

        /// <summary>
        /// Gets or sets likes.
        /// </summary>
        public int Likes { get; set; }

        /// <summary>
        /// Gets or sets dislikes.
        /// </summary>
        public int Dislikes { get; set; }

        /// <summary>
        /// Gets or sets image url.
        /// </summary>
        public string ImageUrl { get; set; }

        /// <summary>
        /// Gets or sets author id.
        /// </summary>
        public string AuthorId { get; set; }

        /// <summary>
        /// Gets or sets application user.
        /// </summary>
        public virtual ApplicationUser Author { get; set; }

        /// <summary>
        /// Gets or sets created at.
        /// </summary>
        public DateTime CreatedAt { get; set; } = DateTime.Now;

        /// <summary>
        /// Gets or sets the comments.
        /// </summary>
        /// <value>
        /// The comments.
        /// </value>
        public virtual ICollection<Comment> Comments { get; set; }

        /// <summary>
        /// Gets or sets the posts tags relations.
        /// </summary>
        /// <value>
        /// The posts tags relations.
        /// </value>
        public virtual ICollection<PostsTagsRelations> PostsTagsRelations { get; set; }
    }
}